src/cascadia/TerminalSettingsModel/MTSMSettings.h

```diff
@@ -96,7 +96,7 @@ Author(s):
 
 #define MTSM_FONT_SETTINGS(X)                                                          \
     X(hstring, FontFace, "face", DEFAULT_FONT_FACE)                                    \
-    X(int32_t, FontSize, "size", DEFAULT_FONT_SIZE)                                    \
+    X(float, FontSize, "size", DEFAULT_FONT_SIZE)                                      \

```

Comment:
Users *hated* when we saved the opacity values with long decimals (`"opacity": .849999999999999999999999999999999999999999999999`).

How can we ensure that we don't mess up their font settings in the same way?

Reply:
We need to prevent conversion between double and float, since that's lossy, and that's basically it. If that still doesn't work it's either a fault in WinUI or because of our `/fp:fast` with the latter _way_ less likely, since that only kicks in if we do calculations on font sizes, but we don't do that when simply storing the font size. Conversion between float and string and back is a lossless operation and can't result in `0.84999999999...` if given `0.85f`.

Reply:
For what it's worth, this happened all the way back to the very early days of us writing JSON files to disk.

It happened regardless of a user setting opacity in the UI, and we did no math on the opacity value at the time. It was a tight feedback loop from JSON parse to JSON serialization!